src/codeu/chat/client/core/View.java

```diff
@@ -136,4 +137,23 @@ public View(ConnectionSource source) {
 
     return messages;
   }
+
+  // Added by Priyanka Agarwal according to CodeU tutorial
+  public ServerInfo getInfo() {
+    try (final Connection connection = this.source.connect()) {
+      Serializers.INTEGER.write(connection.out(), NetworkCode.SERVER_INFO_REQUEST);
+      if (Serializers.INTEGER.read(connection.in()) == NetworkCode.SERVER_INFO_RESPONSE) {
+        final Uuid version = Uuid.SERIALIZER.read(connection.in());
+        return new ServerInfo(version);
+      } else {
+        // Communicate this error - the server did not respond with the type of
```

Comment:
I added comments to explain the function and print lines to display the error information. Please let me know if I can improve this.